sql/core/src/test/scala/org/apache/spark/sql/execution/SparkPlanTest.scala

```diff
@@ -56,6 +60,24 @@ class SparkPlanTest extends SparkFunSuite {
    * @param expectedAnswer the expected result in a [[Seq]] of [[Product]]s.
    */
   protected def checkAnswer[A <: Product : TypeTag](
+      input: DataFrame,
+      planFunction: SparkPlan => SparkPlan,
+      expectedAnswer: Seq[A]): Unit = {
+    val expectedRows = expectedAnswer.map(Row.fromTuple)
+    SparkPlanTest.checkAnswer(input, planFunction, expectedRows) match {
+      case Some(errorMessage) => fail(errorMessage)
+      case None =>
+    }
+  }
+
+  /**
+   * Runs the plan and makes sure the answer matches the expected result.
+   * @param input the input data to be used.
+   * @param planFunction a function which accepts the input SparkPlan and uses it to instantiate
+   *                     the physical operator that's being tested.
+   * @param expectedAnswer the expected result in a [[Seq]] of [[Product]]s.
+   */
+  protected def checkAnswer[A <: Product : TypeTag](
```

Comment:
Yeah, thats probably reasonable, though I'm fine either way since you already did a nice job of extracting all the real work into the `SparkPlanTest` object.  I guess in most cases you probably will want to give names instead of using the defaults though.